File: src/Deviser.Core/Deviser.Core.Common/DomainTypes/ModuleView.cs
﻿using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;

namespace Deviser.Core.Common.DomainTypes
{
    public class ModuleView
    {
        public Guid Id { get; set; }
        public string ActionName { get; set; }
        public string ControllerName { get; set; }
        public string ControllerNamespace { get; set; }
        public string DisplayName { get; set; }
        public string IconImage { get; set; }
        public string IconClass { get; set; }
        public Guid ModuleViewTypeId { get; set; }
        public Guid ModuleId { get; set; }
        public bool IsDefault { get; set; }
        public virtual ModuleViewType ModuleViewType { get; set; }
        public ICollection<Property> Properties { get; set; }
    }
}

File: src/Deviser.Core/Deviser.NgLib/deviser-workspace/projects/deviser-shared/src/lib/domain-types/module-view.ts
import { Property } from './property';
import { ModuleViewType } from './module-view-type';

export interface ModuleView
    {
        id: string;
        actionName: string;
        controllerName: string;
        controllerNamespace: string;
        displayName: string;
        iconImage: string;
        iconClass: string;
        moduleViewTypeId: string;
        moduleId: string;
        isDefault: boolean;
        moduleViewType: ModuleViewType;
        properties: Property[];
    }